File: 2015/01/1.go
package main

import (
	"fmt"
	"io"
	"os"
)

func main() {
	input, err := os.Open("./2015/01/input.txt")
	if err != nil {
		fmt.Fprintln(os.Stderr, "Unable to open input file")
		os.Exit(1)
	}
	var b = make([]byte, 1)
	var floor = 0
	for true {
		_, err := input.Read(b)
		if err != nil {
			if err == io.EOF {
				break
			}
			fmt.Fprintln(os.Stderr, "Unexpected Error when reading input.txt", err)
			os.Exit(1)
		}
		switch string(b) {
		case "(":
			floor++
		case ")":
			floor--
		default:
			fmt.Fprintln(os.Stderr, "Unexpected character '", string(b), "'")
		}
	}

	fmt.Println("Santa will end up on floor ", floor)
}

File: 2015/01/1.py
input = None
try:
    input = open("./2015/01/input.txt", "r")
except:
    print("Error opening input.txt")
    exit(1)

floor = 0

for char in input.read():
    if char == "(":
        floor = floor + 1
    elif char == ")":
        floor = floor - 1
    else:
        print("Warning unknown character: '", char, "'", sep="")

print(floor)
